src/main/kotlin/no/nav/pensjon/kalkulator/simulering/client/pen/PenSimuleringClient.kt

```diff
@@ -40,19 +47,40 @@ class PenSimuleringClient(
         )?.let(PenSimuleringResultMapper::fromDto)
             ?: emptyResult()
 
-    override fun simulerAnonymAlderspensjon(spec: ImpersonalSimuleringSpec) =
-        doPost(
-            path = ANONYM_PATH,
-            requestBody = PenAnonymSimuleringSpecMapper.toDto(spec),
-            requestClass = PenAnonymSimuleringSpec::class.java,
-            responseClass = PenAnonymSimuleringResult::class.java,
-            deprecatedBasePath = true
-        )?.let(PenAnonymSimuleringResultMapper::fromDto)
-            ?: emptyResult()
+
+
+    override fun simulerAnonymAlderspensjon(spec: ImpersonalSimuleringSpec): SimuleringResult {
+        return try {
+            doPost(
+                path = ANONYM_PATH,
+                requestBody = PenAnonymSimuleringSpecMapper.toDto(spec),
+                requestClass = PenAnonymSimuleringSpec::class.java,
+                responseClass = PenAnonymSimuleringResult::class.java,
+                deprecatedBasePath = true
+            )?.let(PenAnonymSimuleringResultMapper::fromDto)
+                ?: emptyResult()
+
+        }  catch (e: EgressException) { //Adding formatted errorObject to the EgressException
+            try {
+                val jsonMap: PenAnonymSimuleringError = mapper.readValue(e.message, PenAnonymSimuleringError::class.java)
+                e.errorObj = jsonMap.let(PenAnonymSimuleringErrorMapper::fromDto)
+            } catch (_: JsonProcessingException) {
+                log.debug { "Json mapping feilet. Ugjenkjennelig format på feilmelding fra PEN: ${e.message}"}
+                e.errorObj =  AnonymSimuleringErrorV1(
+                    status = "PEK500InternalServerError",
+                    message = "Det skjedde en feil i kalkuleringen"
+                )
+            }
```

Comment:
Foreslår å trekke ut denne `try-catch` i en egen funksjon med navn f.eks. "addFormattedErrorObjectToException". Da oppnås to fordeler:

- Mer lettlest kode
- Man slipper å ha kommentarer som "Adding formatted errorObject to the EgressException"